plugins/org.jkiss.dbeaver.registry/src/org/jkiss/dbeaver/registry/EclipseWorkspaceImpl.java

```diff
@@ -130,6 +131,18 @@ protected void loadWorkspaceProjects() throws DBException {
                 if (activeProject == null || (!CommonUtils.isEmpty(activeProjectName) && project.getName().equals(activeProjectName))) {
                     activeProject = projectMetadata;
                 }
+
+                // To avoid accidental corruption of the workspace configuration by search/replace commands,
+                // we need to mark metadata folder as hidden (see dbeaver/dbeaver#20759)
+                IFolder metadataFolder = project.getFolder(DBPProject.METADATA_FOLDER);
+                if (metadataFolder.exists() && !metadataFolder.isHidden()) {
+                    try {
+                        metadataFolder.setHidden(true);
```

Comment:
Hope you tested well that it doesn't affect any functionality of reading and writing to this directory